lib/fluent/plugin/exception_detector.rb

```diff
@@ -59,7 +59,7 @@ def self.supported
     ].freeze
 
     PYTHON_RULES = [
-      rule(:start_state, /Traceback \(most recent call last\)/, :python),
+      rule(:start_state, /^Traceback \(most recent call last\):$/, :python),
```

Comment:
It might be tricky to add similar restrictions for Ruby and Java. The stack trace for Ruby might include a custom error which is not at the beginning of a line. 

```
$ cat error.rb
raise "error"
$ ruby error.rb
error.rb:1:in `<main>': error (RuntimeError)
```

Similarly for java (grabbed from a previous customer ticket https://github.com/GoogleCloudPlatform/fluent-plugin-detect-exceptions/issues/12):
```javax.ws.rs.ProcessingException:
org.apache.http.conn.ConnectTimeoutException: Connect to
ssp-api-stg01.tmadata.co.uk:443
[ssp-api-stg01.tmadata.co.uk/80.195.118.185,
ssp-api-stg01.tmadata.co.uk/217.138.85.11] failed: connect
   at io.dropwizard.client.DropwizardApacheConnector.apply(DropwizardApacheConnector.java:109)
   at org.glassfish.jersey.client.ClientRuntime.invoke(ClientRuntime.java:252)
   at org.glassfish.jersey.client.JerseyInvocation$1.call(JerseyInvocation.java:684)
Caused by: org.apache.http.conn.ConnectTimeoutException: Connect to ssp-api-stg01.tmadata.co.uk:443
[ssp-api-stg01.tmadata.co.uk/80.195.118.185,
ssp-api-stg01.tmadata.co.uk/217.138.85.11] failed: connect timed out
   at org.apache.http.impl.conn.DefaultHttpClientConnectionOperator.connect(DefaultHttpClientConnectionOperator.java:150)
   at org.apache.http.impl.conn.PoolingHttpClientConnectionManager.connect(PoolingHttpClientConnectionManager.java:353)
   at org.apache.http.impl.execchain.MainClientExec.establishRoute(MainClientExec.java:380)
   ... 26 common frames omitted
Caused by: java.net.SocketTimeoutException: connect timed out
   at java.net.AbstractPlainSocketImpl.doConnect(AbstractPlainSocketImpl.java:350)
   at java.net.AbstractPlainSocketImpl.connectToAddress(AbstractPlainSocketImpl.java:207)
   at java.net.AbstractPlainSocketImpl.connect(AbstractPlainSocketImpl.java:188)

   ... 36 common frames omitted
```

We also had users complaining PHP regex was too strict before: https://github.com/GoogleCloudPlatform/fluent-plugin-detect-exceptions/issues/2

It might be nice for us to come up with some customizable regexp at some point.